File: examples/python/templates/addressable_pygen.py
from __future__ import annotations
from typing import *
import examples.templates.addressable as pybind_addressable_pygen
from bind_utils import *
from metaclass import *


class AddressableRoot(metaclass=OriginalMethodsMetaclass):
    """
    Addressable is a base class for all the objects, that have abs path."""
    @bind
    def __init__(self, parent: Root, name: str):
        pass


    
    @bind
    def abs_path(self) -> str:
        """
        comments
        @brief Get object's abs path based on parent's abs path and object's name
        @return the abs path of the object"""
        pass

File: examples/primitives/python/wrappers/templates/addressable_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/templates/addressable.hpp"
#include "cxx/classes/root.hpp"

namespace py = pybind11;

void bindAddressableRoot(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::Addressable<iegen::example::Root>, std::shared_ptr<iegen::example::Addressable<iegen::example::Root>>> addressableRoot(m, "AddressableRoot");

    addressableRoot.def(py::init<const iegen::example::Root &, const std::string &>(), py::arg("parent"), py::arg("name"));
    addressableRoot.def("abs_path", &iegen::example::Addressable<iegen::example::Root>::absPath);
}